presto-main/src/main/java/com/facebook/presto/sql/ExpressionUtils.java

```diff
@@ -258,6 +258,9 @@ public static Expression normalize(Expression expression)
                 ComparisonExpression comparison = (ComparisonExpression) not.getValue();
                 return new ComparisonExpression(negate(comparison.getType()), comparison.getLeft(), comparison.getRight());
```

Comment:
this returns new expression which might not be matched in IdentityHashMap in Analysis (see getType() method).